h2o-algos/src/main/java/hex/glm/DispersionUtils.java

```diff
@@ -151,6 +157,185 @@ public static double estimateTweedieDispersionOnly(GLMModel.GLMParameters parms,
+        Vec mu = gPred.outputFrame(Key.make(), new String[]{"prediction"}, null).vec(0);
+        Vec response = dinfo._adaptedFrame.vec(dinfo.responseChunkId(0));
+        class MomentMethodThetaEstimation extends MRTask<MomentMethodThetaEstimation> {
+            double _muSqSum;
+            double _sSqSum;
+            double _muSum;
+            double _wSum;
+            @Override
+            public void map(Chunk[] cs) {
+                // mu, y, w
+                for (int i = 0; i < cs[0]._len; i++) {
+                    final double w = cs[2].atd(i);
+                    _muSqSum += w * Math.pow(cs[0].atd(i), 2);
+                    _sSqSum += w * Math.pow(cs[1].atd(i) - cs[0].atd(i), 2);
+                    _muSum += w * cs[0].atd(i);
+                    _wSum += w;
+                }
+            }
+
+            @Override
+            public void reduce(MomentMethodThetaEstimation mrt) {
+                _muSqSum += mrt._muSqSum;
+                _sSqSum += mrt._sSqSum;
+                _muSum += mrt._muSum;
+                _wSum += mrt._wSum;
+            }
+        };
+        MomentMethodThetaEstimation mm = new MomentMethodThetaEstimation().doAll(mu, response, weights);
+
+        return mm._muSqSum/(mm._sSqSum - mm._muSum/mm._wSum);
+    }
+
+
+    public static double estimateNegBinomialDispersionFisherScoring(GLMModel.GLMParameters parms, GLMModel model, Job job,
+                                                       double[] beta, DataInfo dinfo) {
+        Vec weights = dinfo._weights
+                ? dinfo.getWeightsVec()
+                : dinfo._adaptedFrame.makeCompatible(new Frame(Vec.makeOne(dinfo._adaptedFrame.numRows())))[0];
+
+        final double nRows = weights == null
+                ? dinfo._adaptedFrame.numRows()
+                : weights.mean() * weights.length();
+
+        DispersionTask.GenPrediction gPred = new DispersionTask.GenPrediction(beta, model, dinfo).doAll(
+                1, Vec.T_NUM, dinfo._adaptedFrame);
+        Vec mu = gPred.outputFrame(Key.make(), new String[]{"prediction"}, null).vec(0);
+        Vec response = dinfo._adaptedFrame.vec(dinfo.responseChunkId(0));
+        double invTheta = nRows / new CalculateInitialTheta().doAll(mu, response, weights)._theta0;
+        double delta = 1;
+        int i = 0;
+        for (; i < parms._max_iterations_dispersion; i++) {
+            if (Math.abs(delta) < parms._dispersion_epsilon) break;
+            invTheta = Math.abs(invTheta);
+            CalculateNegativeBinomialScoreAndInfo si = new CalculateNegativeBinomialScoreAndInfo(invTheta).doAll(mu, response, weights);
+            delta = si._score/si._info;
+            invTheta += delta;
+        }
+
+        if (invTheta < 0)
+            Log.warn("Dispersion estimate truncated at zero.");
+        if (i == parms._max_iterations_dispersion)
+            Log.warn("Iteration limit reached.");
+        return 1./invTheta;
+    }
```

Comment:
Unused. I will remove it unless we want to keep it, e.g., for estimating the dispersion after the model was trained.